src/runtime_src/core/pcie/driver/linux/xocl/subdev/calib_storage.c

```diff
@@ -50,6 +51,9 @@ static int calib_storage_save_by_idx(struct platform_device *pdev, uint32_t idx)
 		goto done;
 	}
 
+	err = xocl_srsr_read_calib(xdev, idx, calib_storage->cache[idx]->data,
+					XOCL_CALIB_CACHE_SIZE);
```

Comment:
why do you have a constant value for XOCL_CALIB_CACHE_SIZE? Shouldn't it be the number of memory banks?

Reply:
calib_storage is the super set of all the calibration cache and make it to an array of struct calib_cache *, each struct calib_cache save 1. mem idx, 2. the 16KB calibration data.

We may need additional field: mem_type to indicate the calibration data belongs to,
just like what we have in mig. 

We can set up a quick call for review the code!

Reply:
done! get cache_size by asking SRSR_IP